bundles/com.espressif.idf.launch.serial.core/src/com/espressif/idf/launch/serial/core/IDFCoreLaunchConfigProvider.java

```diff
@@ -37,6 +38,14 @@ public ILaunchConfiguration getLaunchConfiguration(ILaunchDescriptor descriptor,
 				.findAppropriateLaunchConfig(descriptor, IDFLaunchConstants.RUN_LAUNCH_CONFIG_TYPE) : configuration;
 		configuration = configuration == null ? createLaunchConfiguration(descriptor, target) : configuration;
 		projectConfigs.put(configuration.getName(), configuration);
+
+		String usbLoc = target.getAttribute(IDFLaunchConstants.OPENOCD_USB_LOCATION, (String) null);
+		if (!StringUtil.isEmpty(usbLoc)) {
+			ILaunchConfigurationWorkingCopy wc = configuration.getWorkingCopy();
+			wc.setAttribute(IDFLaunchConstants.OPENOCD_USB_LOCATION, usbLoc);
+			configuration = wc.doSave();
+			projectConfigs.put(configuration.getName(), configuration);
+		}
```

Comment:
I think we can avoid setting this attribute in the launch configuration, since it's already saved in the launch target.

For example, instead of using:
`fLaunchConfiguration.getAttribute(IDFLaunchConstants.OPENOCD_USB_LOCATION, (String) null)`

we can retrieve the active launch target from the launch bar using:
`ILaunchBarManager.getActiveLaunchTarget()`